rocAL/rocAL/source/readers/image/file_source_reader.cpp

```diff
@@ -156,46 +157,47 @@ void FileSourceReader::reset()
 
 Reader::Status FileSourceReader::subfolder_reading()
 {
-    if ((_sub_dir = opendir (_folder_path.c_str())) == nullptr)
-        THROW("FileReader ShardID ["+ TOSTR(_shard_id)+ "] ERROR: Failed opening the directory at " + _folder_path);
-
     std::vector<std::string> entry_name_list;
-    std::string _full_path = _folder_path;
-
-    while((_entity = readdir (_sub_dir)) != nullptr)
-    {
-        std::string entry_name(_entity->d_name);
-        if (strcmp(_entity->d_name, ".") == 0 || strcmp(_entity->d_name, "..") == 0) continue;
-        entry_name_list.push_back(entry_name);
-    }
-    closedir(_sub_dir);
-    std::sort(entry_name_list.begin(), entry_name_list.end());
-
     auto ret = Reader::Status::OK;
-    for (unsigned dir_count = 0; dir_count < entry_name_list.size(); ++dir_count) {
-        std::string subfolder_path = _full_path + "/" + entry_name_list[dir_count];
-        filesys::path pathObj(subfolder_path);
-        if(filesys::exists(pathObj) && filesys::is_regular_file(pathObj))
-        {
-            // ignore files with non-image extensions
-            auto file_extension_idx = subfolder_path.find_last_of(".");
-            if (file_extension_idx  != std::string::npos) {
-                std::string file_extension = subfolder_path.substr(file_extension_idx+1);
-                std::transform(file_extension.begin(), file_extension.end(), file_extension.begin(),
-                    [](unsigned char c){ return std::tolower(c); });
-                if ((file_extension != "jpg") && (file_extension != "jpeg") && (file_extension != "png") && (file_extension != "ppm") && (file_extension != "bmp") && (file_extension != "pgm") && (file_extension != "tif") && (file_extension != "tiff") && (file_extension != "webp"))
+    for (auto& entry : filesys::recursive_directory_iterator(_folder_path.c_str())) {
+        try {
+            std::string entry_path = entry.path().string();
+            auto entry_path_id = entry_path;
+            auto last_slash_idx = entry_path_id.find_last_of("\\/");
+            if (std::string::npos != last_slash_idx)
+            {
+                entry_path_id.erase(0, last_slash_idx + 1);
+            }
+            if (filesys::is_regular_file(entry.path() ))
+            {
+                if(!_meta_data_reader || _meta_data_reader->exists(entry_path_id))
+                {
+                if(get_file_shard_id() != _shard_id )
+                {
+                    _file_count_all_shards++;
+                    incremenet_file_id();
                     continue;
+                }
+                _in_batch_read_count++;
+                _in_batch_read_count = (_in_batch_read_count%_batch_count == 0) ? 0 : _in_batch_read_count;
```

Comment:
Ensure all operations has space before and after

Reply:
done